chainer/functions/math/sparse_matmul.py

```diff
@@ -0,0 +1,512 @@
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+try:
+    from scipy import sparse
+    _scipy_available = True
+except ImportError:
+    _scipy_available = False
+
+
+class SparseCooMatrix(object):
+
+    def __init__(self, data, row, col, shape, use_variable=False):
+        self.data = data
+        if use_variable:
+            self.data = chainer.Variable(self.data)
+        self.row = row
+        self.col = col
+        self.shape = shape  # (row, col)
+
+    def to_dense(self):
+        xp = cuda.get_array_module(self.data)
+        if self.data.ndim == 1:
+            x = xp.zeros(self.shape, dtype=self.data.dtype)
+            x[self.row, self.col] = self.data
+            return x
+        if self.data.ndim == 2:
+            nb = self.data.shape[0]
+            x = xp.zeros((nb, self.shape[0], self.shape[1]),
+                         dtype=self.data.dtype)
+            for i in range(nb):
+                nnz = len(xp.where(self.data[i] != 0)[0])
```

Comment:
How about using `xp.count_nonzero(self.data[i])`

Reply:
Thank you!  That is better.